sample-code/spring-app/src/test/java/com/sap/ai/sdk/app/controllers/GroundingTest.java

```diff
@@ -0,0 +1,146 @@
+    for (final var repository : repositoryList) {
+      assertThat(repository.getId()).isNotNull();
+      assertThat(repository.getTitle()).isNotNull();
+    }
+  }
+
+  @Test
+  void testCreateDeleteCollection() {
+    final var controller = new GroundingController();
+
+    // (1) CREATE COLLECTION
+    final var collectionId = controller.createCollection(JSON_FORMAT);
+    final var collectionUuid = UUID.fromString(collectionId);
+    assertThat(collectionId).isNotNull();
+
+    // (1.1) TEST COLLECTION LOOKUP
+    this.testCollectionsGetAll();
+
+    // (2) SANITY CHECK: NO DOCUMENTS
+    final var documentsEmpty = controller.getDocumentsByCollectionId(collectionUuid, JSON_FORMAT);
+    assertThat(documentsEmpty).isInstanceOf(Documents.class);
+    assertThat(((Documents) documentsEmpty).getCount()).isEqualTo(0);
+    assertThat(((Documents) documentsEmpty).getResources()).isNotNull().isEmpty();
+
+    // (3) UPLOAD A DOCUMENT
+    final var documentsCreated = controller.createDocument(collectionUuid, JSON_FORMAT);
+    assertThat(documentsCreated).isInstanceOf(DocumentsListResponse.class);
+    final var document = ((DocumentsListResponse) documentsCreated).getDocuments();
+    assertThat(document).isNotNull().hasSize(1);
+
+    // (3.1) TEST DOCUMENT LOOKUP
+    this.testGetDocumentById(collectionUuid, document.get(0).getId());
+
+    // (4) SEARCH FOR DOCUMENTS
+    Object search = controller.searchInDocuments(JSON_FORMAT);
+    assertThat(search).isInstanceOf(RetievalSearchResults.class);
+    final var dayOfWeek = now().getDayOfWeek().getDisplayName(TextStyle.FULL, Locale.ENGLISH);
+    this.assertDocumentSearchResult((RetievalSearchResults) search, dayOfWeek);
+
+    // (5) DELETE COLLECTION
+    Object deletion = controller.deleteDocuments(collectionUuid, JSON_FORMAT);
+    assertThat(deletion).isInstanceOf(OpenApiResponse.class);
+    assertThat(((OpenApiResponse) deletion).getStatusCode()).isEqualTo(202);
+
+    // (6) SANITY CHECK: NO COLLECTION
+    assertThatThrownBy(() -> controller.getDocumentsByCollectionId(collectionUuid, JSON_FORMAT))
+        .hasMessageContaining("404 Not Found");
+  }
+
+  private void assertDocumentSearchResult(RetievalSearchResults search, String dayOfWeek) {
+    assertThat(search.getResults()).isNotEmpty();
+    for (final var resultsByFilter : search.getResults()) {
+      assertThat(resultsByFilter.getFilterId()).isEqualTo("question");
+      assertThat(resultsByFilter.getResults()).isNotEmpty();
+      for (final var result : resultsByFilter.getResults()) {
+        assertThat(result.getDataRepository().getDocuments()).isNotEmpty();
+        for (final var document : result.getDataRepository().getDocuments()) {
+          assertThat(document.getChunks()).isNotEmpty();
+          for (final var chunk : document.getChunks()) {
+            assertThat(chunk.getContent()).contains(dayOfWeek);
+          }
+        }
+      }
+    }
+  }
+
+  void testCollectionsGetAll() {
```

Comment:
When will this be tested?

Reply:
It's tested as part of the big test suite `testCreateDeleteCollection()`.

If you prefer, I'm okay with testing it as additional dedicated `@Test` case.